### ISSUE TITLE
full parameter fine tune save_pretrian error when multiple gpu

### ISSUE THREAD

**Issue by dongxiaolong** [axolotl-ai-cloud/axolotl#541]
### Please check that this issue hasn't been reported before.

- [X] I searched previous [Bug Reports](https://github.com/OpenAccess-AI-Collective/axolotl/labels/bug) didn't find any similar reports.

### Expected Behavior

It should be saved as a whole file, not as a tiny error.

### Current behaviour

the file size as below:
`(base) root@autodl-container-ae134bb297-3fcce4a9:~/autodl-tmp/models/CodeLlama-7b-hf-platypus# du -sh ./*
4.0K    ./config.json
4.0K    ./generation_config.json
600K    ./pytorch_model.bin
4.0K    ./special_tokens_map.json
4.0K    ./tokenizer_config.json
492K    ./tokenizer.model`

the train config as below:

base_model: /root/autodl-tmp/models/CodeLlama-7b-hf/
base_model_config: /root/autodl-tmp/models/CodeLlama-7b-hf/
model_type: LlamaForCausalLM
tokenizer_type: CodeLlamaTokenizer
is_llama_derived_model: true

load_in_8bit: true
load_in_4bit: false
strict: false

datasets:
  - path: /root/autodl-tmp/datas/Open-Platypus
    type: alpaca
dataset_prepared_path: /root/autodl-tmp/data/last_run_prepared
val_set_size: 0.01
output_dir: /root/autodl-tmp/models/CodeLlama-7b-hf-platypus

sequence_len: 4096
sample_packing: false

adapter:
lora_model_dir:
lora_r: 32
lora_alpha: 16
lora_dropout: 0.05
lora_target_linear: true
lora_fan_in_fan_out:

wandb_mode: disabled
wandb_project:
wandb_entity:
wandb_watch:
wandb_run_id:
wandb_log_model:

gradient_accumulation_steps: 4
micro_batch_size: 2
num_epochs: 3
optimizer: adamw_bnb_8bit
lr_scheduler: cosine
learning_rate: 0.0002

train_on_inputs: false
group_by_length: false
bf16: true
fp16: false
tf32: false

gradient_checkpointing: true
early_stopping_patience:
resume_from_checkpoint:
local_rank:
logging_steps: 1
xformers_attention:
flash_attention: true

warmup_steps: 10
eval_steps: 20
save_steps: 20
save_total_limit: 2
debug:
deepspeed: deepspeed/zero3.json

ddp_timeout:
ddp_bucket_cap_mb:
ddp_broadcast_buffers:

weight_decay: 0.0
fsdp:
fsdp_config:
special_tokens:
  bos_token: "<s>"
  eos_token: "</s>"
  unk_token: "<unk>"

the trainner log as below:
{'eval_loss': 1.8974876403808594, 'eval_runtime': 51.6082, 'eval_samples_per_second': 4.825, 'eval_steps_per_second': 0.213, 'epoch': 2.95}
 99%|█████████▊| 760/771 [36:52:43<31:32, 172.07s/it]
100%|██████████| 11/11 [00:46<00:00,  4.63s/it]
{'loss': 0.377, 'learning_rate': 0.0002, 'epoch': 2.96}
{'loss': 0.4346, 'learning_rate': 0.0002, 'epoch': 2.96}
{'loss': 0.3881, 'learning_rate': 0.0002, 'epoch': 2.97}
{'loss': 0.3487, 'learning_rate': 0.0002, 'epoch': 2.97}
{'loss': 0.369, 'learning_rate': 0.0002, 'epoch': 2.97}
{'loss': 0.4242, 'learning_rate': 0.0002, 'epoch': 2.98}
{'loss': 0.4793, 'learning_rate': 0.0002, 'epoch': 2.98}
{'loss': 0.3757, 'learning_rate': 0.0002, 'epoch': 2.99}
{'loss': 0.4214, 'learning_rate': 0.0002, 'epoch': 2.99}
{'loss': 0.4014, 'learning_rate': 0.0002, 'epoch': 2.99}
{'loss': 0.3884, 'learning_rate': 0.0002, 'epoch': 3.0}
{'train_runtime': 134656.2344, 'train_samples_per_second': 0.55, 'train_steps_per_second': 0.006, 'train_loss': 0.5674956093038567, 'epoch': 3.0}
100%|██████████| 771/771 [37:24:16<00:00, 172.55s/it][2023-09-08 22:56:28,968] [INFO] [axolotl.train.train:120] [PID:66336] Training Completed!!! Saving pre-trained model to /root/autodl-tmp/models/CodeLlama-7b-hf-platypus
[2023-09-08 22:56:28,968] [INFO] [axolotl.train.train:120] [PID:66334] Training Completed!!! Saving pre-trained model to /root/autodl-tmp/models/CodeLlama-7b-hf-platypus
[2023-09-08 22:56:28,968] [INFO] [axolotl.train.train:120] [PID:66338] Training Completed!!! Saving pre-trained model to /root/autodl-tmp/models/CodeLlama-7b-hf-platypus
[2023-09-08 22:56:28,968] [INFO] [axolotl.train.train:120] [PID:66335] Training Completed!!! Saving pre-trained model to /root/autodl-tmp/models/CodeLlama-7b-hf-platypus
[2023-09-08 22:56:28,968] [INFO] [axolotl.train.train:120] [PID:66343] Training Completed!!! Saving pre-trained model to /root/autodl-tmp/models/CodeLlama-7b-hf-platypus
[2023-09-08 22:56:28,968] [INFO] [axolotl.train.train:120] [PID:66333] Training Completed!!! Saving pre-trained model to /root/autodl-tmp/models/CodeLlama-7b-hf-platypus
[2023-09-08 22:56:28,968] [INFO] [axolotl.train.train:120] [PID:66341] Training Completed!!! Saving pre-trained model to /root/autodl-tmp/models/CodeLlama-7b-hf-platypus
[2023-09-08 22:56:28,968] [INFO] [axolotl.train.train:120] [PID:66342] Training Completed!!! Saving pre-trained model to /root/autodl-tmp/models/CodeLlama-7b-hf-platypus
[2023-09-08 22:56:28,968] [INFO] [axolotl.train.train:120] [PID:66340] Training Completed!!! Saving pre-trained model to /root/autodl-tmp/models/CodeLlama-7b-hf-platypus
[2023-09-08 22:56:28,968] [INFO] [axolotl.train.train:120] [PID:66339] Training Completed!!! Saving pre-trained model to /root/autodl-tmp/models/CodeLlama-7b-hf-platypus
[2023-09-08 22:56:28,968] [INFO] [axolotl.train.train:120] [PID:66337] Training Completed!!! Saving pre-trained model to /root/autodl-tmp/models/CodeLlama-7b-hf-platypus
100%|██████████| 771/771 [37:24:16<00:00, 174.65s/it]
[2023-09-08 22:56:28,969] [INFO] [axolotl.train.train:120] [PID:66332] Training Completed!!! Saving pre-trained model to /root/autodl-tmp/models/CodeLlama-7b-hf-platypus

### Steps to reproduce

1.modify train config as above
2. excute accelerate launch scripts/finetune.py example/codellama/7B/lora.yml

### Possible solution

This could be due to an error caused by a multi-process save.

### Which Operating Systems are you using?

- [X] Linux
- [ ] macOS
- [ ] Windows

### Python Version

3.10

### axolotl branch-commit

main

### Acknowledgements

- [X] My issue title is concise, descriptive, and in title casing.
- [X] I have searched the existing issues to make sure this bug has not been reported yet.
- [X] I am using the latest version of axolotl.
- [X] I have provided enough information for the maintainers to reproduce and diagnose the issue.

**Comment by Kurumella** [comment]
Did you solve this? Facing the same issue during FFT using multi-node/ multi-gpu setup. Full model does not get saved in output_dir

**Comment by winglian** [comment]
@dongxiaolong FFT doesn't work as expected if you set `load_in_8bit: true`, that is only for peft/lora configurations.

**Comment by Kurumella** [comment]
Yes of course, I'm not training in 8bit. My config looks fine and checkpoints are accurate however the final model is incomplete probably due to issues in saving
